rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -49,6 +49,56 @@ SPECIALIZE_GENERIC_C_ARRAY(uint32, uint32_t)
 SPECIALIZE_GENERIC_C_ARRAY(int64, int64_t)
 SPECIALIZE_GENERIC_C_ARRAY(uint64, uint64_t)
 
+// TODO(mikaelarguedas) find right spot to define these
```

Comment:
NOTE: I wanted to implement these arrays through the PRIMITIVE_ARRAY macros [here](https://github.com/ros2/rosidl/blob/master/rosidl_generator_c/include/rosidl_generator_c/primitives_array_functions.h#L29) and [here](https://github.com/ros2/rosidl/blob/master/rosidl_generator_c/include/rosidl_generator_c/primitives_array.h#L22) and the associated c files. Problem is that these macros rely on the result of `sizeof` for memory allocation where here we would like to specify the size of the custom array members via another argument (c.f. l64 parameter `member_size`).
I don't think that adding an optional parameter to all C array definition macros just to cover this specific use case is the way to go.  So I'm considering keeping this `void__Array` definition as a separate piece of code.

I'm open to suggestions regarding which approach makes more sense and, if the latter, where to store these definitions.